### ISSUE TITLE
Make all toView: parameters an optional UIView (for consistency)

### ISSUE THREAD

**Issue by madhikarma** [ustwo/autolayout-helper-swift#2]
### Description

I recently changed to non optionals for certain methods to be non-optional (see https://github.com/ustwo/autolayout-helper-swift/commit/5c09f94e6bb4ba53abacd4ef8b71e9db5dca33c2) but I'm rethinking and wondering if its better that Cocoa crashes inside UIKit if you send in a nil superview instead of it being a no-op (does nothing)
### Solutions
- We could / should change back to optionals also so its easier to integrate and consistent across all methods. Note if you use a method that needs a toView then it will crash when the constraint is created
- Or we could go one step further and make some use default params to reduce method chaining
- Other??

What do you think? @danieladias @chris59david @jamaine-ustwo 

Note. should merge PR #1 first

**Comment by madhikarma** [comment]
e.g.     

```
func addLeftConstraint(toView view: UIView, relation: NSLayoutRelation, constant: CGFloat) -> NSLayoutConstraint {
```

should go back to 

```
func addLeftConstraint(toView view: UIView?, relation: NSLayoutRelation, constant: CGFloat) -> NSLayoutConstraint {
```

and 
        if let superview = label.superview {

```
        label.translatesAutoresizingMaskIntoConstraints = false
        label.addCenterXConstraint(toView: superview, relation: NSLayoutRelation.Equal, constant: 0.0)
        label.addCenterYConstraint(toView: superview, relation: NSLayoutRelation.Equal, constant: 0.0)
    }
```

goes back to 

```
        label.translatesAutoresizingMaskIntoConstraints = false
        label.addCenterXConstraint(toView: label.superview, relation: NSLayoutRelation.Equal, constant: 0.0)
        label.addCenterYConstraint(toView: label.superview, relation: NSLayoutRelation.Equal, constant: 0.0)
```

(same for all public methods so its consistent)

We could also go one step further and use default params (since we now have a dedicated Obj-C version) but i think that could be a separate PR (though I don't mind)

**Comment by danieladias** [comment]
Still unsure crashing is the way to go but if you want this to be more easy to use and devs will get feedback anyway with the crash, it sounds good.

**Comment by madhikarma** [comment]
True thats not very Swift and I hate crashes 

But it might be better to crash if we distribute this is a framework (#2) than mask the problem form the user. 

But I think default params can resolve this from being an issue since you either a) want the view to be there or b) you don't want it and use a constant instead 

Or maybe implicitly unwrapped optionals be used in a parameter? 

I raised this issue as I'm not sure if the recent change I made is worth keeping i.e is it good to cover what UIKit would do if you created that constraint and what if Apple change behaviour under the hood?
